web-common/src/features/dashboards/dimension-table/dimension-table-utils.ts

```diff
@@ -271,3 +215,229 @@ export function estimateColumnSizes(
 
   return estimateColumnSize;
 }
+
+export function prepareVirtualizedDimTableColumns(
```

Comment:
Feels weird having so many arguments. Since most of these are from `MetricsExplorerEntity` we could perhaps get that as an argument?

Reply:
Hey @AdityaHegde, regarding this comment and the other two: I'm doing more work on this stuff to enable sorting by dimension values (have a PR underway right now). Would it be ok to accept this PR for now and I can clean this up as you suggest in the follow-up PR, since I'm making changes to this code anyway?

Reply:
addressed in https://github.com/rilldata/rill/pull/3173/commits/10614a6fefc22766d3941464f9a5dbd829a922b5